site/app/libraries/FileUtils.php

```diff
@@ -765,4 +765,32 @@ public static function getTopEmptyDir(string $path, string $base_path, &$results
             $results[] = $parent_dir;
         }
     }
+
+
+    /**
+     * Given a path, searches for the topmost directory that's empty, here empty means having no files but there can be
+     * empty subdirectories inside that directory
+     *
+     * @param string $path Absolute path to file or directory
+     * @return bool
+     */
+    public static function validPath($path) {
+        $disallowed = [
+            '..',   // Parent directory
+            ':',    // Drive separator (Windows)
+            '*',    // Wildcard character
+            '?',    // Wildcard character
+            '"',    // Double quote
+            '<',    // Less than
+            '>',    // Greater than
+            '|',    // Pipe
+            '\0'    // Null character
+        ];
+        foreach ($disallowed as $char) {
+            if (strpos($path, $char) !== false) {
+                return false;
+            }
+        }
```

Comment:
The file `test..test.out` is a valid filename which does not cause a parent directory issue. Are we now blocking that?